### ISSUE TITLE
False positive TaintedSql on firstWhere() array form: @psalm-taint-sink sql on $column hits array values

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#733]
## Description

`Builder::firstWhere()` has `@psalm-taint-sink sql $column` in its stub, intended to catch column-name injection (`firstWhere($request->input('col'))`). However, when called with an associative array, the **values** inside the array also go into `$column`, which causes a false positive `TaintedSql` even though Eloquent parameterizes all values.

## Stub (Eloquent/Builder.stubphp)

```php
/**
 * @psalm-taint-sink sql $column   ← hits array values too
 * @psalm-taint-escape sql
 * @psalm-flow ($operator, $value) -> return
 * @psalm-taint-specialize
 */
public function firstWhere($column, $operator = null, $value = null, $boolean = 'and') {}
```

## Reproduction

```php
$sid = $request->input('id');   // tainted

// Flagged as TaintedSql — but this is a safe parameterized query
UserToken::firstWhere([
    'driver_id' => $socialite->getId(),  // tainted (Http\Client\Response source)
    'driver'    => $driver,
]);

// Also flagged
Conversation::where(['status_id' => $sid, 'to_id' => $dm->from_id])->delete();
```

Found in real apps:
- `app/Actions/AttemptToAuthenticateSocialite.php` (monica)
- `app/Http/Controllers/DirectMessageController.php:401,407` (pixelfed)
- `app/Http/Controllers/Groups/GroupsCommentController.php:346` (pixelfed)
- `app/Http/Controllers/Groups/GroupsPostController.php:351` (pixelfed)

## Root Cause

When `firstWhere(['key' => $taintedValue])` is called:

- The array (including its tainted values) goes into the `$column` parameter
- `@psalm-taint-sink sql $column` fires because `$column` received tainted data
- Eloquent's actual SQL execution is fully parameterized — no injection risk

Compare with `where()`, which has **no** `@psalm-taint-sink sql $column` and never false-positives on the array form:

```php
// where() stub — correct, no false positives on array form
* @psalm-taint-escape sql
* @psalm-flow ($value) -> return
* @psalm-taint-specialize
```

## Fix

Remove `@psalm-taint-sink sql $column` from `firstWhere()`, making it consistent with `where()`:

```php
/**
 * @psalm-taint-escape sql
 * @psalm-flow ($operator, $value) -> return
 * @psalm-taint-specialize
 */
public function firstWhere($column, $operator = null, $value = null, $boolean = 'and') {}
```

The column-name injection risk in `firstWhere()` is the same as in `where()` — both accept a raw column string in position 1, and neither is typically called with user-supplied column names. The actual column-name injection danger (`orderBy($col)`) is correctly covered by a separate sink annotation on `orderBy()`.

**Comment by alies-dev** [comment]
Closing as duplicate of #734. Same root cause (`@psalm-taint-sink sql $column` fires on array VALUES in the array form); #734 already cross-references this issue and covers the fix for both Eloquent `firstWhere()` and the QueryBuilder `where/orWhere/whereNot/orWhereNot/having/orHaving` family.
